packages/legend-extension-dsl-data-space/src/components/query/DataSpaceQueryBuilderTemplateQueryPanelContent.tsx

```diff
@@ -0,0 +1,192 @@
+              vertical: 'top',
+              horizontal: 'center',
+            }}
+          >
+            <div className="query-builder__data-space__template-query-panel">
+              <div className="query-builder__data-space__template-query-panel__header">
+                Curated Template Queries
+              </div>
+              {templateQueries.map((query) => (
+                <div
+                  key={query.title}
+                  className="query-builder__data-space__template-query-panel__query"
+                >
+                  <TagIcon className="query-builder__data-space__template-query-panel__query__icon" />
+                  <button
+                    className="query-builder__data-space__template-query-panel__query__entry"
+                    title="click to load template query"
+                    onClick={() => loadQuery(query)}
+                  >
+                    <div className="query-builder__data-space__template-query-panel__query__entry__content">
+                      <div className="query-builder__data-space__template-query-panel__query__entry__content__title">
+                        {query.title}
+                      </div>
+                      {query.description && (
+                        <div className="query-builder__data-space__template-query-panel__query__entry__content__description">
+                          {query.description}
+                        </div>
+                      )}
+                    </div>
+                  </button>
+                  <button
+                    className="query-builder__data-space__template-query-panel__query__share"
+                    title="Share..."
+                    onClick={() => shareTemplateQuery(query)}
+                  >
+                    <ShareIcon />
+                    <div className="query-builder__data-space__template-query-panel__query__share__label">
+                      Share
+                    </div>
+                  </button>
+                </div>
+              ))}
+            </div>
+          </BasePopover>
+        </div>
+      </ClickAwayListener>
+    );
+  },
+);
+
+const DataSpaceQueryBuilderTemplateQueryPanel = observer(
+  (props: { queryBuilderState: DataSpaceQueryBuilderState }) => {
+    const { queryBuilderState } = props;
+    const templateQueryButtonRef = useRef<HTMLButtonElement>(null);
+    const templateQueries = queryBuilderState.dataSpace.executables?.filter(
+      (e) => e instanceof DataSpaceExecutableTemplate,
+    ) as DataSpaceExecutableTemplate[];
+
+    const showTemplateQueries = (): void => {
+      queryBuilderState.setTemplateQueryDialogOpen(true);
+    };
+
+    return (
+      <PanelHeader className="query-builder__data-space__template-query">
+        <div className="query-builder__data-space__template-query__title">
+          <FilterIcon />
+        </div>
+        <button
+          className="query-builder__data-space__template-query__btn"
```

Comment:
is there any benefit to showing this if there is no template queries in the dataSpace? 
Seems like that would just take up valuable UI space right?
could we hide if there are no templates?